src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -959,7 +959,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           doIt
         }
 
-
+        def isJavaOrUniversal(sym: Symbol) = {
+          sym.isJavaDefined ||
+          isUniversalMember(sym) ||
+          (if (sym.name.endsWith("$extension")) AnyTpe.nonPrivateMember(sym.name.dropRight(10)) != NoSymbol else false)
```

Comment:
Can we avoid this by running `checkCanAutoApply` only if `!phase.erasedTypes`?

Reply:
`!phase.erasedTypes` didn't work but the following did:

```scala
val afterExtMethods = phaseId(currentPeriod) >= currentRun.phaseNamed("extmethods").id
```

and then `!afterExtMethods`.